quic/s2n-quic-transport/src/space/datagram.rs

```diff
@@ -38,48 +40,101 @@ impl<Config: endpoint::Config> Manager<Config> {
     ) {
         let mut packet = Packet {
             context,
-            pending_streams: stream_manager.pending_streams(),
+            has_pending_streams: stream_manager.has_pending_streams(),
         };
         self.sender.on_transmit(&mut packet);
     }
 }
+
+impl<Config: endpoint::Config> interest::Provider for Manager<Config> {
+    #[inline]
+    fn transmission_interest<Q: interest::Query>(&self, query: &mut Q) -> interest::Result {
+        if self.sender.has_transmission_interest() {
+            query.on_new_data()?;
+        }
+        Ok(())
+    }
+}
+
 struct Packet<'a, C: WriteContext> {
     context: &'a mut C,
-    pending_streams: bool,
+    has_pending_streams: bool,
 }
 
 const FRAME_TYPE_LEN: usize = 1;
+const MAX_LEN_VALUE: usize = 8;
 
 impl<'a, C: WriteContext> s2n_quic_core::datagram::Packet for Packet<'a, C> {
     /// Returns the remaining space in the packet
     fn remaining_capacity(&self) -> usize {
         let space = self.context.remaining_capacity();
-        //= https://www.rfc-editor.org/rfc/rfc9221#section-4
-        //# The least significant bit of the Type field in the DATAGRAM frame is
-        //# the LEN bit (0x01), which indicates whether there is a Length field
-        //# present: if this bit is set to 0, the Length field is absent and the
-        //# Datagram Data field extends to the end of the packet; if this bit is
-        //# set to 1, the Length field is present.
-        // In the case where the user writes the largest datagram possible
-        // we don't factor in the size of the Length field as it will be the last
-        // frame in the packet. Thus we only have to factor out the frame type field.
-        space - FRAME_TYPE_LEN
+        // Remove the frame type length and the maximum length value
+        space
+            .saturating_sub(FRAME_TYPE_LEN)
+            .saturating_sub(MAX_LEN_VALUE)
```

Comment:
I think it would be better to ask the encoding size of each of the values, rather than hard-coding them:

```suggestion
            .saturating_sub(frame::datagram:DATAGRAM_TAG.encoding_size())
            .saturating_sub(VarInt::new(space).unwrap_or(VarInt::MAX).encoding_size())
```